### PR TITLE
Add 1d support for ArborXWrappers

### PR DESCRIPTION
**Pull request by masterleinad**
Related to https://github.com/dealii/dealii/pull/15538.

### CODE REVIEW

**Comment by Rombur** [comment]
We now require ArborX 1.3 which has multi-dimensional search support. It's probably cleaner and faster to use the new interface instead of copying everything to 3D data structures.

**Review by tamiko — APPROVED** [review]

**Comment by masterleinad** [comment]
> We now require ArborX 1.3 which has multi-dimensional search support. It's probably cleaner and faster to use the new interface instead of copying everything to 3D data structures.

That requires incompatible changes to the predicates, though, since they are not templated on the dimension but store 3-dim objects. I would rather do that in a follow-up.

**Comment by Rombur** [comment]
Using CTAD,  adding a template on the dimension should be backward compatible because the constructor of the predicates takes a `Point` or a `BoundingBox` that is already templated on the dimension.

File: source/arborx/access_traits.cc [hunk 1]
```diff
@@ -26,16 +26,15 @@ namespace ArborXWrappers
...
     for (unsigned int i = 0; i < size; ++i)
       {
         points.emplace_back(static_cast<float>(dim_points[i][0]),
-                            static_cast<float>(dim_points[i][1]),
-                            dim == 2 ? 0.f :
-                                       static_cast<float>(dim_points[i][2]));
+                            dim < 2 ? 0.f :
+                                      static_cast<float>(dim_points[i][1]),
+                            dim < 3 ? 0.f :
+                                      static_cast<float>(dim_points[i][2]));
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply source/arborx/access_traits.cc]: See [`c8b16f5` (#15720)](https://github.com/dealii/dealii/pull/15720/commits/c8b16f593c7c8677488cf6015a3e9e6826b45177).

File: source/arborx/access_traits.cc [hunk 2]
```diff
@@ -21,22 +21,31 @@ DEAL_II_NAMESPACE_OPEN
...
+    dealii::Point<3, float>
+    to_3d_float_point(dealii::Point<dim, Number> p)
+    {
+      static_assert(dim >= 1 && dim <= 3);
+      if (dim == 1)
+        return {float(p[0]), 0.f, 0.f};
+      else if (dim == 2)
+        return {float(p[0]), float(p[1]), 0.f};
+      else
+        return {float(p[0]), float(p[1]), float(p[2])};
```
Comment (bangerth) [inline source/arborx/access_traits.cc:36]: I don't think you actually need all of these explicit conversions. `Point<3,Number>` has a constructor that takes three `Number` objects -- which means the conversion can be implicit:
```suggestion
      if (dim == 1)
        return Point<3,float>(p[0], 0.f, 0.f);
      else if (dim == 2)
        return Point<3,float>(p[0], p[1], 0.f);
      else
        return Point<3,float>(p[0], p[1], p[2]);
```

**Review by Rombur — APPROVED** [review]

**Comment by Rombur** [comment]
clang-format is not happy

**Review by tamiko — APPROVED** [review]

**Pull request merged**